### ISSUE TITLE
Add an optional lock to nodes in the hierarchy

### ISSUE THREAD

**Issue by LaraEngebretson** [AtomicGameEngine/AtomicGameEngine#358]
This is to prevent them from changing, moving, or getting deleted. This would also need to be visible in the inspector panel, i.e. the settings could be grayed out or the blue button indicating that a node is open/closed could become red.

Originally requested here: #159

**Comment by Alan-FGR** [comment]
What about a 'static' tickbox to mark objects that don't move so we can use that information for all kinds of optimizations besides the practical implications in workflow of course?

**Comment by mattbenic** [comment]
That sounds useful, but the initial feature that just locks them from editor changes would be worthwhile as well.

**Comment by Alan-FGR** [comment]
Yep, locking stuff in the editor, and disabling selection perhaps for some layers would be useful for sure.
